### DISCUSSION TITLE
Prevent undoInputRule

### DISCUSSION THREAD

**Discussion by mc-martin-o** [ueberdosis/tiptap#722] [General Questions]
Hi,
currently I am working with InputRules to add german typographical quotes which are a bit different to the english ones. But there is one thing I would like to prevent and that is the logic for the backspace key after the rule is applied. (Does not make sense for one character replacements at least for me) I found that binding backspace to the undoInputRule happens in the setup for the Editor class. Is there any way to override that behaviour?
Regards Martin.

**Comment by fridaystreet** [comment]
Also looking for how to do this, I'm converting text based emojis into their actual emojis, but on backspace rather than remove the entire emoji, it just reverts to the text version and all chars need to be deleted

**Comment by fridaystreet** [comment]
Update: found a bit of an answer and just needed to work through how to use it for our use case, but basically you just override the backspace keyboard shortcut. Just takes a bit of working out as to what my work for the particular node you're removing

**Comment by space-squad** [comment]
> currently I am working with InputRules to add german typographical quotes

@mc-martin-o could you picture sharing your code for this? This would help us a ton :)

**Comment by mc-martin-o** [comment]
@space-squad 
The extension looks like this...
```
export default Extension.create({
  name: 'smartquotes',

  addOptions() {
    return {
      startQuoteChar: '„',
      endQuoteChar: '“',
    };
  },

  addInputRules() {
    return [
      textInputRule({
        find: /(?:[\w\d.,!?)\]}„])(")$/,
        replace: this.options.endQuoteChar,
      }),
      textInputRule({
        find: /"$/,
        replace: this.options.startQuoteChar,
      }),
    ];
  },
});
```

**Reply by space-squad** [reply]
Thank you very much! Works like a charm

**Reply by eriknygren** [reply]
@mc-martin-o, I see you had a discussion over at [Prosemirror](https://github.com/ProseMirror/prosemirror-inputrules/commit/8b6eabc2c2a6d737ac33e5080d5607e0a25b77a5) and managed to get an `undoable` option added for input rules. Did you figure out a way to pipe that through into tiptap? Working on the very same thing around smart quotes! Thanks a lot.

**Reply by mc-martin-o** [reply]
@eriknygren I didn't even know that there is this new option and am still using prosemirror-inputrules 1.2.0 🫣.
Thanks for pointing out to me.

**Reply by eriknygren** [reply]
You're welcome lol! Let me know if you figure something out please, I got stuck with this and instead I'm doing some post-processing of text to replace quote symbols now which is far from ideal. Would love to do it the "official" way.
